samples/mqtt/pkcs11_connect/main.cpp

```diff
@@ -199,99 +179,24 @@ int main(int argc, char *argv[])
     fprintf(stdout, "Connecting...\n");
     if (!connection->Connect(clientId.c_str(), false /*cleanSession*/, 1000 /*keepAliveTimeSecs*/))
     {
-        fprintf(stderr, "Failed to kick off MQTT async Connect(): %s\n", ErrorDebugString(connection->LastError()));
+        fprintf(stderr, "MQTT Connection failed with error %s\n", ErrorDebugString(connection->LastError()));
         exit(-1);
     }
 
-    // wait for the OnConnectionCompleted callback to fire, which sets connectionCompletedPromise...
-    if (connectionCompletedPromise.get_future().get() == false)
+    if (connectionCompletedPromise.get_future().get())
     {
-        fprintf(stderr, "Connection failed\n");
-        exit(-1);
-    }
+        /* Wait for half a second */
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
why bother waiting?

my worry in having this here is that customers are going to copy/paste it because they'll think it's important